### PR TITLE
Refining interfaces page

### PR DESCRIPTION
**Pull request by co9olguy**

### CODE REVIEW

File: doc/introduction/interfaces.rst [hunk 1]
```diff
@@ -6,12 +6,33 @@
 Interfaces
 ==========
 
-PennyLane integrates quantum nodes made up of variational circuits with other programming
-and machine learning frameworks.
-Such frameworks are called *interfaces*. The default interface, implicitly used in the previous section on
-how to program :ref:`variational circuits <intro_vcircuits>`, is NumPy.
+PennyLane offers seamless integration between classical and quantum computations. Code up quantum 
+circuits in PennyLane, and connect them easily to the top scientific computing and machine learning 
+libraries.
```
Comment (josh146) [inline doc/introduction/interfaces.rst]: This is a really nice intro. Should steal this for the features list 😆

File: doc/introduction/interfaces.rst [hunk 2]
```diff
@@ -6,12 +6,33 @@
...
 
-Currently, there is support for the following three interfaces:
+The bridge between the quantum and classical worlds is provided in PennyLane via *interfaces*. 
+Currently, there are three built-in interfaces: NumPy, PyTorch, and TensorFlow. 
+These interfaces make each of these libraries quantum-aware, allowing quantum circuits to be
+treated just like any other operation. 
+
+In PennyLane, an interface is declared when creating a :class:`~.QNode`, e.g.,
```
Comment (josh146) [inline doc/introduction/interfaces.rst]: Does the QNode link work?

**Comment by josh146** [comment]
Note: code coverage went down only because the TF 1.X tests failed on Travis (looks like the container lost internet connection). Restarted the job, that should fix it.

File: doc/introduction/interfaces.rst (hunk 2 above)
Comment (co9olguy) [reply doc/introduction/interfaces.rst]: locally, yes (but still some issues with css on the navbar)

**Pull request merged**